### PR TITLE
fix: Update Netlify SSR link

### PR DESCRIPTION
**Pull request by jacobthesheep**
## Changes

- depends on https://github.com/withastro/docs/pull/4859

## Testing

N/A

## Docs

/cc @withastro/maintainers-docs for feedback!
- https://github.com/withastro/docs/pull/4859

### CODE REVIEW

**Comment by jacobthesheep** [comment]
> I know why you changed the link, however using the new link I do not get moved to the heading, while the old link still moves me to the `Adapter for SSR/Edge`. Maybe worth double checking?

Well that's because the another PR hasn't been merged.

**Comment by ematipico** [comment]
@jacobthesheep could you address the comment in the other PR? It seems there are broken links. As you said, we can't merge this PR if the other isn't merged.

**Review by sarah11918 — APPROVED** [review]
Confirming! This change is what the docs PR needs. It'll all work out, and the docs one will be merged and live before this one carries over. Docs has control over when this PR merges into docs, so we can merge in the proper order over there. No harm in merging here.

**Pull request merged**
